src/main/python/hydra/zmqtest/runtest.py

```diff
@@ -26,42 +28,111 @@ def __init__(self, argv):
             config_fn = argv[1]
             del argv[1]
         RunTestBase.__init__(self, 'zmqScale', config, config_fn)
-
-        zstpub = '/zst-pub'
-        zstsub = '/zst-sub'
-        self.add_appid(zstpub)
-        self.add_appid(zstsub)
+        self.zstpub = '/zst-pub'
+        self.zstsub = '/zst-sub'
+        self.add_appid(self.zstpub)
+        self.add_appid(self.zstsub)
+        #self.total_sub_apps = 200
+        self.total_sub_apps = zmq_config.total_apps
         self.start_init()
+        self.run_test()
+
+    def run_test(self):
+        # Launch zmq pub
+        self.launch_zmq_pub()
+
+        # Launch zmq sub up to self.total_sub_apps
+        self.launch_zmq_sub()
+
+        # Signal PUB to start sending all messages, blocks until PUB notifies
+        l.info("Waiting for 10 secs before signalling PUB to send messages ..")
+        time.sleep(10)
+        self.signal_pub_send_msgs()
+
+        #  Fetch all client SUB data
+        self.fetch_all_sub_clients_data()
+
+        #  Delete all launched apps
+        self.delete_all_launched_apps()
+        l.info("Successfully finished gathering all data")
+
+        l.info("================================================")
+        for client, info in self.all_sub_clients_info.items():
+            l.info("Client: %s", client)
```

Comment:
printing valid results are not useful here, just print if an error is detected i.e. some packet are lost.